scene/2d/animated_sprite.cpp

```diff
@@ -353,18 +353,31 @@ void AnimatedSprite::_notification(int p_what) {
 
 					timeout=1.0/speed;
 
-					int fc = frames->get_frame_count(animation);
-					if (frame>=fc-1) {
-						if (frames->get_animation_loop(animation)) {
-							frame=0;
-						} else {
-							frame=fc-1;
-						}
-					} else {
-						frame++;
-						if (frame==fc-1) {
-							emit_signal(SceneStringNames::get_singleton()->finished);
-						}
+					const int fc = frames->get_frame_count(animation);
+					
+					// nextframe
+					int nextFrame = frame + 1;
+					if (frames->get_animation_loop(animation)) {
```

Comment:
This seems bogus, it means looping animations would always get either `0` or `fc - 1` as `nextFrame`.

Overall the rewritten logic in the second commit makes it harder to read IMO.